### DISCUSSION TITLE
洛谷宣传贴内容

### DISCUSSION THREAD

**Discussion by boomzero** [langningchen/XMOJ-Script#190] [Ideas]
因为只能发一次, 所以还是想好再发吧

**Comment by boomzero** [comment]
### 介绍

XMOJ增强脚本是一个功能强大的浏览器扩展，旨在提升用户在XMOJ（ http://www.xmoj.tech ）上的使用体验。
该脚本通过添加多种实用功能和界面优化，使用户能够更高效地参与竞赛、管理代码和享受更好的界面交互。

- **比赛ACM排名与下载功能**：允许用户查看比赛的 ACM 排名，并提供下载选项，方便离线查阅。
- **恢复讨论区和短消息**：允恢复讨论区和短消息, 方便用户沟通。
- **用户名颜色和 Rating**：~让 XMOJ 充分洛谷化~。
- **查看更多标程**：展示更多的标准程序代码，帮助用户更好地理解题目要求和正确解法。
- **获取别人的测试点数据**：允许用户获取其他人的测试点数据，用于分析问题和优化代码。
- **自动刷新比赛列表与排名**：使比赛列表和排名页面自动定时刷新，获取最新信息。
- **检测是否抄袭当年代码**：检测是否抄袭以前写的代码, 并在 ACM 排名中显示。
- **界面时间自动倒计时**：在比赛列表等界面上自动显示剩余时间倒计时，方便用户了解时间情况。
- **回放视频下载功能**：为回放视频添加下载选项，方便用户离线观看回放。
- **自动提交已AC题目**：自动将已通过的题目再次提交，提高题目的通过率。
- **自动选择O2优化**：在代码提交界面自动选择O2优化选项，提高代码效率。
- **界面美化**：改善界面外观，提供更好的用户体验。
    - 使用新版的Bootstrap样式库：更新界面样式，使之更现代。
    - 重新排版：对界面进行重新排版，提高可读性。
    - 增加彩色文字：为界面增加彩色文字元素，增添视觉吸引力。
    - 状态界面添加单位：为状态界面的内存和耗时添加单位，方便理解。
    - 暗色模式：引入暗色模式，减少眼部疲劳。
    - 增加动画：为界面元素增加动画效果，提升交互体验。
    - 对错符号替换：将题目前的对错符号替换为勾和叉，更直观。
    - 去除多余提示：消除界面上多余反复的提示信息。
    - 中文翻译：统一使用中文，翻译部分英文内容，提高可理解性。
    - 替换链接为按钮：将方括号包装的链接替换为按钮，提升操作便捷性。
    - 移除无用功能：去除无法使用的功能，精简界面。
    - 关键词替换：将特定关键词替换为相应内容，增强可读性。
- **自动跳转到登录界面**：在需要登录的页面自动跳转到登录界面，提高操作便捷性。
- **自动保存用户名与密码**：自动保存登录信息，免去每次手动输入的步骤。
- **复制测试样例**：在题目界面复制测试样例内容，方便粘贴到本地编辑器。
- **自动刷新状态页面结果**：使状态页面中的结果自动刷新，便于实时查看变化。
- **复制题目或题解内容**：允许用户复制题目或题解的内容，方便保存和分享。
- **一键打开所有题目**：在比赛题目界面一键打开所有题目，提高操作效率。
- **代码检查功能**：在提交代码前对代码进行检查，以确保代码质量。
    - 是否使用了文件输入输出：检查是否正确使用了文件输入输出功能。
    - 是否有编译错误：检查是否存在编译错误，避免提交无效代码。
- **修复登录后跳转失败**：解决登录后跳转问题，确保用户能够顺利进入下一步操作。
- **下载页面增加内容**：在下载页面增加更多内容，提供更多资源下载。
- **比较代码**：允许用户比较不同版本的代码，方便进行代码对比与分析。

**Comment by boomzero** [comment]
改了readme, 加了新功能

**Comment by boomzero** [comment]
**登录后跳转失败**

**Reply by boomzero** [reply]
这是feature?

**Comment by boomzero** [comment]
@langningchen

**Comment by langningchen** [comment]
很好

**Comment by langningchen** [comment]
我就直接编辑你的那一条了。

**Comment by langningchen** [comment]
实际上，那些功能说明和前面后面的简介都是我把代码里面功能列表给ChatGPT，让他帮我生成的

**Reply by boomzero** [reply]
...

**Reply by boomzero** [reply]
那我用它写一个广告

**Reply by langningchen** [reply]
可以

**Comment by langningchen** [comment]
@boomzero 怎么样？发了吗？

**Reply by boomzero** [reply]
@langningchen 我在写作业...

~我不会告诉你我作业写不完了~

**Comment by PythonSmall-Q** [comment]
要不我来发？

**Comment by PythonSmall-Q** [comment]
顺手发一下我的论坛（？

**Comment by boomzero** [comment]
我发吧

**Reply by langningchen** [reply]
:+1:

**Comment by boomzero** [comment]
你也可以发一个

**Reply by langningchen** [reply]
不用了，谢谢

**Comment by boomzero** [comment]
@langningchen 你的洛谷号是什么, ~互关一下~

**Reply by langningchen** [reply]
等你发完以后我写qp你就知道了

**Comment by PythonSmall-Q** [comment]
https://xmoj.flarum.cloud/blog/41
我发在我的论坛上了

**Comment by PythonSmall-Q** [comment]
置顶。快去下面转转

**Comment by PythonSmall-Q** [comment]
@boomzero @langningchen 你们要什么牌子？看一下https://xmoj.flarum.cloud/d/33-guan-li-yuan-ming-dan/

**Reply by boomzero** [reply]
~admin~

**Reply by PythonSmall-Q** [reply]
我给你咕咕咕

**Reply by PythonSmall-Q** [reply]
牌子好了

**Reply by langningchen** [reply]
我呢？

**Comment by boomzero** [comment]
**XMOJ-Script**是一款专为XMOJ平台设计的浏览器插件，它可以为您提供许多便捷和实用的功能，让您在XMOJ上的编程体验更加舒适和高效。以下是一些XMOJ-Script的亮点功能：

- **比赛ACM排名，并且能下载ACM排名**：您可以在比赛页面查看ACM排名，并且可以下载排名数据，方便您分析和总结。
- **恢复讨论与短消息功能(包括管理, @, 置顶....基于CLoudflare workers)**：您可以在XMOJ上与其他用户进行讨论和交流，支持管理、@、置顶等功能，基于Cloudflare workers，速度快，安全可靠。
- **查看到更多标程**：您可以在题目页面查看更多的标程，学习不同的解题思路和技巧。
- **添加用户评分和用户名颜色**：您可以在用户页面查看用户的评分，并且可以根据评分给用户名添加不同的颜色，增加视觉效果。~让 XMOJ 充分洛谷化~
- **获取到别人的测试点数据**：您可以在状态页面查看别人的测试点数据，帮助您发现和修复错误。
- **比赛列表、比赛排名界面自动刷新**：您可以在比赛列表和比赛排名界面自动刷新数据，实时了解最新的情况。
- **比赛列表等界面的时间自动倒计时**：您可以在比赛列表等界面看到时间自动倒计时，方便您安排时间和策略。
- **回放视频增加下载功能**：您可以在回放视频页面下载视频文件，方便您离线观看和回顾。
- **自动提交已AC题目以提高AC率**：您可以在设置中开启自动提交已AC题目的功能，以提高您的AC率，展示您的实力。
- **代码提交界面自动选择O2优化**：您可以在代码提交界面自动选择O2优化，提高代码的运行效率和性能。

除了以上这些功能，XMOJ-Script还有更多的功能等待您去发现和体验，例如美化界面、自动登录、保存密码、复制题目或题解内容、一键打开所有题目、提交代码前对代码进行检查、导出AC代码、比较代码等等。XMOJ-Script还支持调试模式，方便开发者进行测试和修改。

XMOJ-Script是一款免费、开源、安全的浏览器插件，它只需要很少的内存和网络资源，就可以为您带来极大的便利和乐趣。如果您想了解更多关于XMOJ-Script的信息，或者想下载和安装XMOJ-Script，请访问[官方网站](https://web.xmoj-bbs.tech)或者[GitHub仓库](https://github.com/langningchen/XMOJ-Script/)。

感谢您对XMOJ-Script的关注和支持，希望您在XMOJ上享受编程的乐趣！

**Comment by boomzero** [comment]
how about this?

**Reply by boomzero** [reply]
@langningchen

**Reply by langningchen** [reply]
好的

**Reply by PythonSmall-Q** [reply]
@langningchen 你要啥徽章？

**Comment by langningchen** [comment]
@boomzero 发了吗？

**Comment by boomzero** [comment]
马上发

**Comment by boomzero** [comment]
https://www.luogu.com.cn/discuss/675243

**Comment by boomzero** [comment]
@langningchen

**Comment by langningchen** [comment]
OK

**Comment by langningchen** [comment]
有人问要不要举报……

**Reply by boomzero** [reply]
不要

**Comment by langningchen** [comment]
我很无语

**Comment by boomzero** [comment]
@ 一点xmoj上的人吧

**Comment by langningchen** [comment]
我不知道他们的用户

**Comment by langningchen** [comment]
话说你认识 [KobeBeanBryantCox](https://www.luogu.com.cn/user/865625) 吗？

**Reply by boomzero** [reply]
no

**Reply by langningchen** [reply]
wangjianbo?

**Comment by boomzero** [comment]
@sean_coder 一下

**Comment by langningchen** [comment]
你来

**Reply by boomzero** [reply]
...

**Reply by boomzero** [reply]
不敢

**Reply by langningchen** [reply]
那就散了

**Reply by boomzero** [reply]
@ 了

**Reply by langningchen** [reply]
好的

**Comment by langningchen** [comment]
> @[chy01](https://www.luogu.com.cn/user/856326) 2023-08-29 13:16
> 怎么注册

快来回答他的好问题 @boomzero

**Reply by boomzero** [reply]
没法注册...

**Reply by boomzero** [reply]
除非直接找高老师

**Reply by boomzero** [reply]
但是 ~不能做广告~

**Reply by langningchen** [reply]
......

**Reply by langningchen** [reply]
那就别理他了

**Reply by langningchen** [reply]
@[aaaadou](https://www.luogu.com.cn/user/550935)  2023-08-29 13:25[回复](javascript: scrollToId('replyarea'))举报
这是啥 OJ？

**Reply by boomzero** [reply]
小明的OJ

**Reply by langningchen** [reply]
那你回他吗？……

**Reply by boomzero** [reply]
done

**Comment by langningchen** [comment]
@boomzero 你怎么把我们的Discuss截图放上去了……？

**Reply by boomzero** [reply]
表达对他的无语

**Reply by langningchen** [reply]
好吧

**Comment by langningchen** [comment]
你给@[aaaadou](https://www.luogu.com.cn/user/550935)的回复真厉害

**Comment by boomzero** [comment]
不过没人回 `谁问你了` 已经很好了

**Reply by langningchen** [reply]
确实

**Comment by boomzero** [comment]
我markdown炸了, 链接放错了

**Reply by langningchen** [reply]
第一个错了

**Reply by boomzero** [reply]
就是 xmoj.tech 那个

**Comment by langningchen** [comment]
@[aaaadou](https://www.luogu.com.cn/user/550935)  2023-08-29 13:37[回复](javascript: scrollToId('replyarea'))举报
@[__zcr](https://www.luogu.com.cn/user/773335) 能简单介绍一下吗（为什么连注册窗口都没有……）

**Reply by boomzero** [reply]
你回...

**Reply by langningchen** [reply]
为什么……？

**Reply by langningchen** [reply]
他是 @ 你

**Reply by boomzero** [reply]
@langningchen

**Reply by langningchen** [reply]
？

**Comment by langningchen** [comment]
我刚才按了一下保存讨论

**Reply by boomzero** [reply]
+1

**Comment by boomzero** [comment]
> 他是 @ 你

你写的啊

**Reply by langningchen** [reply]
我写的脚本但是这个帖子是你发的

**Reply by langningchen** [reply]
如果没有人注意都不知道 langningc2009 这个洛谷账号是脚本开发者的

**Reply by boomzero** [reply]
done

**Reply by langningchen** [reply]
你的回复很正确

**Reply by boomzero** [reply]
done

**Comment by langningchen** [comment]
宣传完了，Close discussion

**Comment by PythonSmall-Q** [comment]
我看到了hh

**Reply by langningchen** [reply]
怎么样？ @boomzero 厉害吧？

**Reply by PythonSmall-Q** [reply]
哈哈

**Reply by PythonSmall-Q** [reply]
你们两个的牌子都弄好了

**Reply by PythonSmall-Q** [reply]
去看看吧

**Reply by langningchen** [reply]
@boomzero 你以后别改我的Comment行吧？

**Reply by langningchen** [reply]
> 去看看吧

@PythonSmall-Q 很好

**Reply by PythonSmall-Q** [reply]
https://xmoj.flarum.cloud/d/35-geng-xin-ri-zhi

**Reply by PythonSmall-Q** [reply]
加了个2FA认证累死人了

**Reply by langningchen** [reply]
@PythonSmall-Q 你关了验证码那么 https://github.com/PythonSmall-Q/autolog 是不是就行了？

**Reply by PythonSmall-Q** [reply]
那个**玩意儿是自带的。关不了

**Reply by langningchen** [reply]
好吧

**Comment by boomzero** [comment]
@langningchen 

...
![](https://github.com/langningchen/XMOJ-Script/assets/85378277/1f9fc3c0-e5a4-40b9-9537-ae61d06907f6)

**Comment by langningchen** [comment]
@boomzero ......
